docs/source/en/kv_cache.md

```diff
@@ -0,0 +1,222 @@
+    messages = [{"role": "user", "content": "Hello, what's your name."}]
+    inputs = tokenizer.apply_chat_template(messages, add_generation_prompt=True, return_tensors="pt", return_dict=True).to(model.device)
+    generated_ids = inputs.input_ids
+    max_new_tokens = 10
+    for _ in range(max_new_tokens):
+        outputs = model(**inputs, past_key_values=past_key_values, use_cache=True)     
+        # Greedily sample one next token
+        next_token_ids = outputs.logits[:, -1:].argmax(-1)
+        generated_ids = torch.cat([generated_ids, next_token_ids], dim=-1)   
+        # Prepare inputs for the next generation step by leaaving unprocessed tokens, in our case we have only one new token
+        # and expanding attn mask for the new token, as explained above
+        attention_mask = torch.cat(inputs.attention_mask)
+        attention_mask = torch.cat([attention_mask, attention_mask.new_ones((attention_mask.shape[0], 1))], dim=-1)
+        inputs = {"input_ids": next_token_ids, "attention_mask": attention_mask}
+    print(tokenizer.batch_decode(generated_ids, skip_special_tokens=True))
+    </code></pre>
+</details>
+
+
+
+## Generate with Cache
+
+In 🤗 Transformers, we support various Cache types to optimize the performance across different models and tasks. By default all models generate with caching
+and for that we use the ["DynamicCache"] class. It allows us to dynamically grow cache size, by saving more and more keys and values as we generate. Usually all you have to do to enable or disable it is to pass in `use_cache=True` or `use_cache=False` into the `generate()` method.
+
+Apart from that we support many cache classes, which can be set with a `cache_implementation` argument. It can be set to one of `["static", "quantized", "sliding_window", "hybrid"]`. Let's look into each cache type in more details.
+
+### Quantized Cache
+
+The key and value cache can occupy a large portion of memory, becoming a bottleneck for long-context generation, especially for Large Language Models.
+Quantizing the cache when using `generate()` can significantly reduce memory requirements at the cost of speed.
+
+KV Cache quantization in `transformers` is largely inspired by the paper [KIVI: A Tuning-Free Asymmetric 2bit Quantization for KV Cache]
+(https://arxiv.org/abs/2402.02750) and currently supports `quanto` and `HQQ` as backends. For more information on the inner workings see the paper.
+
+To enable quantization of the key-value cache, one needs to indicate `cache_implementation="quantized"` in the `generation_config`.
+Quantization related arguments should be passed to the `generation_config` either as a `dict` or an instance of a [`QuantizedCacheConfig`] class.
+One has to indicate which quantization backend to use in the [`QuantizedCacheConfig`], the default is `quanto`.
+
+<Tip warning={true}>
+
+Cache quantization can be detrimental if the context length is short and there is enough GPU VRAM available to run without cache quantization.
```

Comment:
I know this is copied but... it's detrimental in what sense? Let's be more precise :D